### ISSUE TITLE
Compiling 'SSL_get_verify_callback@@OPENSSL_1.0.0' error

### ISSUE THREAD

**Issue by DeeGee78** [sumoprojects/sumokoin#58]
Hi guys

Trying to compile and keep getting to 77% and it fails with the error below. Did some research and found some recommendations to try make LIBS="-lcrypto"  but still same issue. Please advise how I can resolve?

undefined reference to symbol 'SSL_get_verify_callback@@OPENSSL_1.0.0'
//lib/x86_64-linux-gnu/libssl.so.1.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/wallet/CMakeFiles/wallet_rpc_server.dir/build.make:127: recipe for target 'bin/sumo-wallet-rpc' failed
make[3]: *** [bin/sumo-wallet-rpc] Error 1
make[3]: Leaving directory '/home/podm/Sumo/build/release'
CMakeFiles/Makefile2:1161: recipe for target 'src/wallet/CMakeFiles/wallet_rpc_server.dir/all' failed
make[2]: *** [src/wallet/CMakeFiles/wallet_rpc_server.dir/all] Error 2
make[2]: Leaving directory '/home/podm/Sumo/build/release'
Makefile:138: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/podm/Sumo/build/release'
Makefile:43: recipe for target 'release' failed
make: *** [release] Error 2

**Comment by paulie-g** [comment]
Seeing the same issue on current Arch x64.

**Comment by Ossii** [comment]
Same here on ubuntu 16.04.
```
[ 78%] Linking CXX executable ../../bin/sumo-wallet-rpc
/usr/bin/ld: /tmp/ccjeCEu7.ltrans19.ltrans.o: undefined reference to symbol 'SSL_get_verify_callback@@OPENSSL_1.0.0'
//lib/x86_64-linux-gnu/libssl.so.1.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/wallet/CMakeFiles/wallet_rpc_server.dir/build.make:128: recipe for target 'bin/sumo-wallet-rpc' failed
make[3]: *** [bin/sumo-wallet-rpc] Error 1
make[3]: Leaving directory '/home/sumopool/temp/sumokoin/build/release'
CMakeFiles/Makefile2:1129: recipe for target 'src/wallet/CMakeFiles/wallet_rpc_server.dir/all' failed
make[2]: *** [src/wallet/CMakeFiles/wallet_rpc_server.dir/all] Error 2
make[2]: Leaving directory '/home/sumopool/temp/sumokoin/build/release'
Makefile:138: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/sumopool/temp/sumokoin/build/release'
Makefile:43: recipe for target 'release' failed
make: *** [release] Error 2
```

**Comment by gamaraan** [comment]
Getting the same error. I tried both Ubuntu 16.04.3 and Arch Linux.

**Comment by SadBatman** [comment]
How do you compile it ? With make static-release ?

**Comment by Ossii** [comment]
cd sumokoin
  make

Is how I did it. Though, I ended up running all this on my debian stretch box and there it was no problem to compile it as above.
Wild guess, gcc-5? :)

**Comment by gamaraan** [comment]
I've made it to compile succesfully using
`make CXXFLAGS="-ldl -lssl -lcrypto"`

**Comment by KingOfLinux** [comment]
Compiled using:
`make CXXFLAGS="-ldl -lssl -lcrypto"`
On Arch Linux getting 'SSL_set_verify@@OPENSSL_1_1_0' error as well.

**Comment by quangvu3** [comment]
Can you guys try with the latest code?

**Comment by Irlasos** [comment]
I tried with Ubuntu 5.4.0-6ubuntu1~16.04.5

I got the same error but after it reach 100%:

`make[3]: Entering directory '/home/poney/sumokoin/build/release'
[ 97%] Building CXX object src/simplewallet/CMakeFiles/simplewallet.dir/simplewallet.cpp.o
make[3]: Leaving directory '/home/poney/sumokoin/build/release'
[ 97%] Built target blockchain_import
[ 98%] Linking CXX executable ../../bin/sumokoind
make[3]: Leaving directory '/home/poney/sumokoin/build/release'
[ 98%] Built target blockchain_export
[ 99%] Linking CXX executable ../../bin/sumo-wallet-rpc
[100%] Linking CXX executable ../../bin/sumo-wallet-cli
/usr/bin/ld: /tmp/ccenkGOC.ltrans20.ltrans.o: undefined reference to symbol 'SSL_get_verify_callback@@OPENSSL_1.0.0'
//lib/x86_64-linux-gnu/libssl.so.1.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/simplewallet/CMakeFiles/simplewallet.dir/build.make:127: recipe for target 'bin/sumo-wallet-cli' failed
make[3]: *** [bin/sumo-wallet-cli] Error 1
make[3]: Leaving directory '/home/poney/sumokoin/build/release'
CMakeFiles/Makefile2:1387: recipe for target 'src/simplewallet/CMakeFiles/simplewallet.dir/all' failed
make[2]: *** [src/simplewallet/CMakeFiles/simplewallet.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
/usr/bin/ld: /tmp/ccz1Z78k.ltrans22.ltrans.o: undefined reference to symbol 'SSL_get_verify_callback@@OPENSSL_1.0.0'
//lib/x86_64-linux-gnu/libssl.so.1.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/daemon/CMakeFiles/daemon.dir/build.make:263: recipe for target 'bin/sumokoind' failed
make[3]: *** [bin/sumokoind] Error 1
make[3]: Leaving directory '/home/poney/sumokoin/build/release'
CMakeFiles/Makefile2:1550: recipe for target 'src/daemon/CMakeFiles/daemon.dir/all' failed
make[2]: *** [src/daemon/CMakeFiles/daemon.dir/all] Error 2
/usr/bin/ld: /tmp/ccCquAJ0.ltrans18.ltrans.o: undefined reference to symbol 'SSL_get_verify_callback@@OPENSSL_1.0.0'
//lib/x86_64-linux-gnu/libssl.so.1.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/wallet/CMakeFiles/wallet_rpc_server.dir/build.make:127: recipe for target 'bin/sumo-wallet-rpc' failed
make[3]: *** [bin/sumo-wallet-rpc] Error 1
make[3]: Leaving directory '/home/poney/sumokoin/build/release'
CMakeFiles/Makefile2:1129: recipe for target 'src/wallet/CMakeFiles/wallet_rpc_server.dir/all' failed
make[2]: *** [src/wallet/CMakeFiles/wallet_rpc_server.dir/all] Error 2
make[2]: Leaving directory '/home/poney/sumokoin/build/release'
Makefile:138: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/poney/sumokoin/build/release'
Makefile:43: recipe for target 'release' failed
make: *** [release] Error 2
`

**Comment by Warlordas** [comment]
Same here with the latest git clone

/usr/bin/ld: /tmp/ccQOIved.ltrans18.ltrans.o: undefined reference to symbol 'SSL_get_verify_callback@@OPENSSL_1.0.0'
//lib/x86_64-linux-gnu/libssl.so.1.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/wallet/CMakeFiles/wallet_rpc_server.dir/build.make:127: recipe for target 'bin/sumo-wallet-rpc' failed
make[3]: *** [bin/sumo-wallet-rpc] Error 1

**Comment by jpenner123** [comment]
Workaround:

try using
make release-static

Worked for me.

**Comment by sumoprojects** [comment]
Fixed with PR #80
